### PR TITLE
Add base of jammy

### PR DESCRIPTION
**Pull request by shayancanonical**
# Issue
1. We have no way of running our suite of integration tests against multiple releases of ubuntu
2. We are not currently supporting `jammy`
3. `ops_test.build_charm` does not support building multiple charm bases

# Solution
1. Use Github CI matrix to run each integration test against a list of `series`. Additonally, use `pytest_generate_test` to add the `series` passed from `tox` to into each test.
2. Add support for `jammy`
3. Manually run `charmcraft pack` in a subprocess to pack charms and then pick the correct archive file to deploy based on the series

# Release Notes
Add base for jammy

### CODE REVIEW

**Review by paulomach — APPROVED** [review]
lgtm - consider pulling main, since there was some test fixes

File: conftest.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+def pytest_addoption(parser):
+    parser.addoption("--series", action="store", type=str, default="jammy")
+
+
+def pytest_generate_tests(metafunc):
```
Comment (WRFitch) [inline conftest.py:10]: This is neat, I like this

**Review by WRFitch — APPROVED** [review]

**Review by marceloneppel — APPROVED** [review]
LGTM! Very good idea on how to generate tests for both series!

**Comment by shayancanonical** [comment]
Thanks for the reviews!

**Pull request merged**